pkg/controller/rdconfig.go

```diff
@@ -21,11 +21,146 @@ import (
 	"os"
 	"reflect"
 
-	rdConfigv1 "github.com/noironetworks/aci-containers/pkg/rdconfig/apis/aci.snat/v1"
+	rdConfig "github.com/noironetworks/aci-containers/pkg/rdconfig/apis/aci.snat/v1"
+	rdconfigclset "github.com/noironetworks/aci-containers/pkg/rdconfig/clientset/versioned"
+	"github.com/noironetworks/aci-containers/pkg/util"
 	apierrors "k8s.io/apimachinery/pkg/api/errors"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/fields"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/watch"
+	"k8s.io/client-go/tools/cache"
 )
 
+func (cont *AciController) initRdConfigInformerFromClient(
+	rdConfigClient *rdconfigclset.Clientset) {
+	ns := os.Getenv("ACI_SNAT_NAMESPACE")
+	name := os.Getenv("ACI_RDCONFIG_NAME")
+	cont.initRdConfigInformerBase(
+		&cache.ListWatch{
+			ListFunc: func(options metav1.ListOptions) (runtime.Object, error) {
+				options.FieldSelector = fields.Set{"metadata.name": name}.String()
+				obj, err := rdConfigClient.AciV1().RdConfigs(ns).List(context.TODO(), options)
+				if err != nil {
+					cont.log.Fatalf("Failed to list RdConfigs during initialization of RdConfigInformer with err: %v", err)
+				}
+				return obj, err
+			},
+			WatchFunc: func(options metav1.ListOptions) (watch.Interface, error) {
+				options.FieldSelector = fields.Set{"metadata.name": name}.String()
+				obj, err := rdConfigClient.AciV1().RdConfigs(ns).Watch(context.TODO(), options)
+				if err != nil {
+					cont.log.Fatalf("Failed to watch RdConfigs during initialization RdConfigsInformer with err: %v", err)
+				}
+				return obj, err
+			},
+		})
+}
+
+func (cont *AciController) initRdConfigInformerBase(listWatch *cache.ListWatch) {
+	cont.rdConfigIndexer, cont.rdConfigInformer = cache.NewIndexerInformer(
+		listWatch,
+		&rdConfig.RdConfig{}, 0,
+		cache.ResourceEventHandlerFuncs{
+			AddFunc: func(obj interface{}) {
+				cont.RdConfigAdded(obj)
+			},
+			UpdateFunc: func(oldiobj interface{}, newobj interface{}) {
+				cont.RdConfigUpdated(oldiobj, newobj)
+			},
+			DeleteFunc: func(obj interface{}) {
+				cont.RdConfigDeleted(obj)
+			},
+		},
+		cache.Indexers{cache.NamespaceIndex: cache.MetaNamespaceIndexFunc},
+	)
+	cont.log.Debug("Initializing RdConfig Informers")
+}
+
+func (cont *AciController) RdConfigAdded(obj interface{}) {
+	rdcon := obj.(*rdConfig.RdConfig)
+	rdconkey, err := cache.MetaNamespaceKeyFunc(rdcon)
+	if err != nil {
+		return
+	}
+	cont.log.Infof("RdConfig: %s added", rdconkey)
+	cont.indexMutex.Lock()
+	cont.rdConfigCache[rdcon.ObjectMeta.Name] = rdcon
+	cont.indexMutex.Unlock()
+	cont.queueRdConfigUpdateByKey(rdconkey)
+}
+
+func (cont *AciController) RdConfigUpdated(oldobj interface{}, newobj interface{}) {
+	oldrdcon := oldobj.(*rdConfig.RdConfig)
+	newrdcon := newobj.(*rdConfig.RdConfig)
+	rdconkey, err := cache.MetaNamespaceKeyFunc(newrdcon)
+	if err != nil {
+		return
```

Comment:
Please log the error (Could not create key) before returning

Reply:
Same here. void return